setup.sh

```diff
@@ -1,3 +1,5 @@
+set -e
```

Comment:
Ensures the bash script returns an error code if any of the individual lines inside it return an error. This was added because if CMake couldn't find ALSA it would silently fail and let you attempt to compile. With this added it errors at the first stage and tells you to add whatever it couldn't find.